jupyterlabutils/notebook/efd_helper.py

```diff
@@ -0,0 +1,142 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import aioinflux
+from collections import Counter
+import logging
+import pandas as pd
+
+from jupyterlabutils.notebook import NotebookAuth
+
+
+class EfdClient:
+    """Class to handle connections and basic queries"""
+    def __init__(self, efd_name, db_name='efd', port='443', path_to_creds='~/.lsst/notebook_auth.yaml'):
+        self.db_name = db_name
+        self.auth = NotebookAuth(path=path_to_creds)
+        self.host, self.user, self.password = self.auth.get_auth(efd_name)
+        self.client = aioinflux.InfluxDBClient(host=self.host,
+                                               port=port,
+                                               ssl=True,
+                                               username=self.user,
+                                               password=self.password,
+                                               db=db_name,
+                                               mode='async')  # mode='blocking')
+        self.client.output = 'dataframe'
+
+    async def get_topics(self):
+        topics = await self.client.query('SHOW MEASUREMENTS')
+        return topics['name'].tolist()
+
+    async def get_fields(self, topic_name):
+        fields = await self.client.query(f'SHOW FIELD KEYS FROM "{self.db_name}"."autogen"."{topic_name}"')
+        return fields['fieldKey'].tolist()
+
+    async def select_time_series(self, topic_name, fields, t1, t2, is_window=False):
+        """Select a time series for a set of topics in a single subsystem"""
+        if not t1.tz:
+            raise ValueError('No timezone information found.  Timezone must be set.')
+        if not t1.tz.zone == 'UTC':
+            logging.warn('Timestamps must be in UTC.  Converting...')
+            t1 = t1.tz_convert(tz='UTC')
+
+        if not isinstance(t1, pd.Timestamp):
+            raise TypeError('The first time argument must be a time stamp')
+        if isinstance(t2, pd.Timestamp):
+            t2 = t2.tz_convert(tz='UTC')
+            start = t1.isoformat()
+            end = t2.isoformat()
+        elif isinstance(t2, pd.Timedelta):
+            if is_window:
+                start = (t1 - t2/2).isoformat()
+                end = (t1 + t2/2).isoformat()
+            else:
+                start = t1.isoformat()
+                end = (t1 + t2).isoformat()
+        else:
+            raise TypeError('The second time argument must be the time stamp for the end ' +
+                            'or a time delta.')
+
+        timespan = f"time >= '{start}' AND time <= '{end}'"
+
+        if isinstance(fields, str):
```

Comment:
I think this sections could be shortened by doing:
```
if isinstance(fields, bytes):
    fields = fields.decode()
fields = [fields, ]
```

Reply:
I want the user to be able to pass a list of fields.  The above would result in a single element list of lists in that situation, so I think I have to do the explicit check for both `str` and `bytes`.